src/tools/ceph_monstore_tool.cc

```diff
@@ -701,6 +701,83 @@ int rebuild_monstore(const char* progname,
   return 0;
 }
 
+void get_name_by_pid(pid_t pid, char *task_name)
+{
+  char proc_pid_path[PATH_MAX] = {0};
+  snprintf(proc_pid_path, PATH_MAX, "/proc/%d/cmdline", pid);
+  int fd = open(proc_pid_path, O_RDONLY);
+
+  if (fd < 0) {
+    fd = -errno;
+    std::cerr << "Fail to open '" << proc_pid_path
+              << "' error = " << cpp_strerror(fd)
+              << std::endl;
+    return;
+  }
+
+  int ret = read(fd, task_name, PATH_MAX-1);
+  if (ret < 0) {
+    ret = -errno;
+    std::cerr << "Fail to read '" << proc_pid_path
+              << "' error = " << cpp_strerror(ret)
+              << std::endl;
+  }
+
+  close(fd);
+}
+
+
+int check_lock(const string &path) {
+ 
+  string::const_reverse_iterator rit;
+  int pos = 0;
+  for (rit = path.rbegin(); rit != path.rend(); ++rit, ++pos) {
+    if (*rit != '/')
+      break;
+  }
+  ostringstream os;
+  os << path.substr(0, path.size() - pos) << "/store.db/LOCK";
+  string lock_path = os.str();
+
+  //The lock file is not existed.
+  if (access(lock_path.c_str(), F_OK)) {
+     return 0;
+  }
+
+  int fd = open(lock_path.c_str(), O_RDONLY);
```

Comment:
this `fd` is never closed in the function. While this is fine if the program exits to user-space, it feels weird and is definitely not ideal if our checks are successful (and thus we continue our work).